packages/ra-core/src/auth/Authenticated.spec.tsx

```diff
@@ -1,24 +1,41 @@
 import * as React from 'react';
 import expect from 'expect';
-import { waitFor } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { Provider } from 'react-redux';
 
-import Authenticated from './Authenticated';
-import AuthContext from './AuthContext';
-import { renderWithRedux } from 'ra-test';
-import { showNotification } from '../actions/notificationActions';
 import { createMemoryHistory } from 'history';
-import { Router } from 'react-router-dom';
+import { Routes, Route, useLocation } from 'react-router-dom';
+import { CoreAdminContext, createAdminStore } from '../core';
+import { showNotification } from '../actions';
+import Authenticated from './Authenticated';
 
 describe('<Authenticated>', () => {
     const Foo = () => <div>Foo</div>;
 
     it('should render its child by default', async () => {
-        const { dispatch, queryByText } = renderWithRedux(
-            <Authenticated>
-                <Foo />
-            </Authenticated>
+        const authProvider = {
+            login: () => Promise.reject('bad method'),
+            logout: () => Promise.reject('bad method'),
+            checkAuth: jest.fn().mockResolvedValueOnce(''),
+            checkError: () => Promise.reject('bad method'),
+            getPermissions: () => Promise.reject('bad method'),
+        };
+        const store = createAdminStore();
+        const dispatch = jest.spyOn(store, 'dispatch');
+
+        render(
+            <Provider store={store}>
+                <CoreAdminContext
+                    authProvider={authProvider}
+                    dataProvider={() => Promise.resolve()}
+                >
+                    <Authenticated>
+                        <Foo />
+                    </Authenticated>
+                </CoreAdminContext>
+            </Provider>
         );
-        expect(queryByText('Foo')).not.toBeNull();
+        expect(screen.queryByText('Foo')).not.toBeNull();
```

Comment:
The rest of this test is wrong. We're testing that redux isn't called... which tests a side effect (the logout notification) of what we actually want to test. 

In fact, the test should be as follows:

1. render the component
2. see that it's content is rendered
3. trigger the authProvider.checkauth response
4. see that the conetn is still rendered

Reply:
No, we're testing that the child is rendered even though we have not called the authProvider yet

Reply:
As discussed, I didn't change the original tests. Testing more things si out of scope for this PR